src/main/java/seedu/address/logic/commands/session/EditSessionCommand.java

```diff
@@ -72,17 +72,21 @@ public CommandResult execute(Model model) throws CommandException {
         Session sessionToEdit = lastShownSessionList.get(index.getZeroBased());
         Session editedSession = createEditedSession(sessionToEdit, editSessionDescriptor);
 
+        // Checks for duplicate sessions
         if (!sessionToEdit.isIdentical(editedSession) && model.hasSession(editedSession)) {
             throw new CommandException(MESSAGE_DUPLICATE_SESSION);
         }
 
+        // Checks for overlapping sessions
         for (int i = 0; i < sessionList.size(); i++) {
```

Comment:
i think for this part it is also possible to use 
`for (Session session : sessionList)`